File: wagerproof-ios-native/WagerproofKit/Sources/WagerproofStores/NBAGameSheetStore.swift
import Foundation
import Observation
import WagerproofModels

/// Drives the NBA game detail bottom sheet. Mirrors the RN
/// `NBAGameSheetContext` (provider + `useNBAGameSheet` hook) but uses
/// SwiftUI's `.sheet(item:)` pattern — when `selectedGame != nil`, the sheet
/// is presented; nil-ing it dismisses.
///
/// `openGameSheet(_:)` matches RN's API. `closeGameSheet()` is provided for
/// imperative dismissal (e.g. when the sheet's close button calls it).
@Observable
@MainActor
public final class NBAGameSheetStore {
    public var selectedGame: NBAGame?

    public init() {}

    public func openGameSheet(_ game: NBAGame) {
        selectedGame = game
    }

    public func closeGameSheet() {
        selectedGame = nil
    }
}

File: wagerproof-android-native/core/stores/src/main/kotlin/com/wagerproof/core/stores/NBAGameSheetStore.kt
package com.wagerproof.core.stores

import androidx.compose.runtime.Stable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.setValue
import com.wagerproof.core.models.NBAGame

/**
 * Drives the NBA game detail bottom sheet. Mirrors the iOS `NBAGameSheetStore`
 * (RN `NBAGameSheetContext` + `useNBAGameSheet`) — `selectedGame != nil`
 * presents a [androidx.compose.material3.ModalBottomSheet]; nil-ing it
 * dismisses. See [NFLGameSheetStore] — identical shape, different model.
 */
@Stable
class NBAGameSheetStore {
    var selectedGame by mutableStateOf<NBAGame?>(null)

    fun openGameSheet(game: NBAGame) {
        selectedGame = game
    }

    fun closeGameSheet() {
        selectedGame = null
    }
}
